src/Components/Popover.tsx

```diff
@@ -0,0 +1,164 @@
+import { isFunction } from "lodash"
+import React from "react"
+import ReactDOM from "react-dom"
+import sizeMe from "react-sizeme"
+import styled, { StyledFunction } from "styled-components"
+import { fadeIn, fadeOut } from "../Assets/Animations"
+import { secondary as secondaryFont } from "../Assets/Fonts"
+
+interface Props extends React.HTMLProps<HTMLDivElement> {
+  show?: boolean
+  minWidth?: string
+  anchorRef?: HTMLElement
+  dismissOnClick?: boolean
+  message?: (() => any) | string
+}
+
+interface State {
+  anchorPosition?: any
+  show: boolean
+  message: any
+}
+
+export class Popover extends React.Component<Props, State> {
+  public messageRef
+
+  static defaultProps = {
+    show: false,
+    dismissOnClick: true,
+  }
+
+  constructor(props) {
+    super(props)
+    const message = this.getMessage(props.message)
+    const show = props.show && message
+    const anchorPosition = this.getAnchorPosition(props.anchorRef)
+    this.state = {
+      show,
+      message,
+      anchorPosition,
+    }
+  }
+
+  componentWillReceiveProps(nextProps: Props) {
+    const newMessage = this.getMessage(nextProps.message)
+    const show = nextProps.show && newMessage
+    const anchorPosition = this.getAnchorPosition(nextProps.anchorRef)
+    this.setState({
+      show,
+      message: newMessage,
+      anchorPosition,
+    })
+  }
+
+  componentDidMount() {
+    window.addEventListener("scroll", this.handleScroll)
```

Comment:
This means the `getAnchorPosition` gets called and the `setState` possibly re-renders this component on every scroll event, which may not be as performant as it should.